sky/provision/kubernetes/instance.py

```diff
@@ -0,0 +1,713 @@
+            '  else '
+            f'    ( echo {exceptions.INSUFFICIENT_PRIVILEGES_CODE!r}; ); '
+            '  fi; '
+            'fi')
+    ]
+
+    for new_node in new_nodes:
+        privilege_check = run_command_on_pods(new_node.metadata.name, namespace,
+                                              check_k8s_user_sudo_cmd)
+        if privilege_check == str(exceptions.INSUFFICIENT_PRIVILEGES_CODE):
+            raise config_lib.KubernetesError(
+                'Insufficient system privileges detected. '
+                'Ensure the default user has root access or '
+                '"sudo" is installed and the user is added to the sudoers '
+                'from the image.')
+
+
+def _setup_ssh_in_pods(namespace: str, new_nodes: List) -> None:
+    # Setting up ssh for the pod instance. This is already setup for
+    # the jump pod so it does not need to be run for it.
+    set_k8s_ssh_cmd = [
+        '/bin/sh',
+        '-c',
+        (
+            'prefix_cmd() '
+            '{ if [ $(id -u) -ne 0 ]; then echo "sudo"; else echo ""; fi; }; '
+            'export DEBIAN_FRONTEND=noninteractive;'
+            '$(prefix_cmd) apt-get update;'
+            '$(prefix_cmd) apt install openssh-server rsync -y; '
+            '$(prefix_cmd) mkdir -p /var/run/sshd; '
+            '$(prefix_cmd) '
+            'sed -i "s/PermitRootLogin prohibit-password/PermitRootLogin yes/" '
+            '/etc/ssh/sshd_config; '
+            '$(prefix_cmd) sed '
+            '"s@session\\s*required\\s*pam_loginuid.so@session optional '
+            'pam_loginuid.so@g" -i /etc/pam.d/sshd; '
+            'cd /etc/ssh/ && $(prefix_cmd) ssh-keygen -A; '
+            '$(prefix_cmd) mkdir -p ~/.ssh; '
+            '$(prefix_cmd) cp /etc/secret-volume/ssh-publickey '
+            '~/.ssh/authorized_keys; '
+            '$(prefix_cmd) chown -R $(whoami) ~/.ssh;'
+            '$(prefix_cmd) chmod 700 ~/.ssh; '
+            '$(prefix_cmd) chmod 600 ~/.ssh/authorized_keys; '
+            '$(prefix_cmd) service ssh restart; '
+            # Eliminate the error
+            # `mesg: ttyname failed: inappropriate ioctl for device`.
+            # See https://www.educative.io/answers/error-mesg-ttyname-failed-inappropriate-ioctl-for-device  # pylint: disable=line-too-long
+            '$(prefix_cmd) sed -i "s/mesg n/tty -s \\&\\& mesg n/" ~/.profile;')
+    ]
+    # TODO(romilb): We need logging and surface errors here.
+    for new_node in new_nodes:
+        run_command_on_pods(new_node.metadata.name, namespace, set_k8s_ssh_cmd)
+
+
+def _label_pod(namespace: str, pod_name: str, label: Dict[str, str]) -> None:
+    """Label a pod."""
+    kubernetes.core_api().patch_namespaced_pod(
+        pod_name,
+        namespace, {'metadata': {
+            'labels': label
+        }},
+        _request_timeout=kubernetes.API_TIMEOUT)
+
+
+def _setup_service(namespace: str, service_spec: Optional[Dict[str, Any]],
+                   pods: List) -> None:
```

Comment:
Is this method required? I thought all services are setup in `config.py::_configure_services` and `service_spec` here seems to be None..

Reply:
It was directly adopted from the `node_provider`. Should we remove it?
https://github.com/skypilot-org/skypilot/blob/553082297ff17b100fb0d61c4a70e5fef4d77c2c/sky/skylet/providers/kubernetes/node_provider.py#L409-L421

Reply:
If you think it is ok to remove them, I am happy to do so. : )

Reply:
ahh looks like it is no longer needed (was a remnant of old YAML format), I think it can be removed